recipes/moc/all/conanfile.py

```diff
@@ -0,0 +1,121 @@
+        ("Macos", "clang", "6")
+    ]
+    name = "moc"
+    homepage = "https://www.github.com/zuut/moc"
+    description = "Moc, the marked-up object compiler"
+    topics = ("conan", "moc", "idl", "code generation")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"    
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = { "fPIC": [True, False] }
+    default_options = { "fPIC": True }
+    _cmake = None
+
+    @property
+    def _version(self):
+        if hasattr(self, "version") and self.version != None:
+            return self.version
+        if hasattr(self, "_default_version") and self._default_version != None:
+            self.version = self._default_version
+            return self.version
+        vmax="0"
+        lvmax = LooseVersion(vmax)
+        for v in self.conan_data["sources"]:
+            lv = LooseVersion(v)
+            try:
+                if lv > lvmax:
+                    vmax = v
+                    lvmax = LooseVersion(vmax)
+            except:
+                self.output.error("unable to compare %s to %s" %(lv,lvmax))
+        self.version = vmax
+        return self.version
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if not self._check_compiler():
+            raise ConanInvalidConfiguration("%s package is not compatible with os %s and compiler %s version %s." % (self.name, self.settings.os, self.settings.compiler, self.settings.compiler.version))
+
+    def _check_compiler(self):
+        os = self.settings.os
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(compiler.version)
+        return any(os == sc[0] and compiler == sc[1] and compiler_version >= sc[2] for sc in self._supported_compilers)
+
+    def build_requirements(self):
+        self.build_requires("flex/2.6.4")
+        self.build_requires("bison/3.7.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self._version])
+        rename(self.name + "-" + self._version, self._source_subfolder)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        if self._noCmakeConfigFiles:
+            for file in glob(self.package_folder + "/lib/**/*-config.cmake") :
+                move(file, file + "-sample")
```

Comment:
The hooks complain about *-config.cmake which are installed by the cmake install. I could remove them instead of renaming them. I renamed them to assist debugging. This was a very painful process for a newbie.